airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -100,58 +105,173 @@ private void saveAttemptSyncConfig(final long jobId, final int attemptNumber, fi
         "set attempt sync config");
   }
 
+  private IntegrationLauncherConfig getSourceIntegrationLauncherConfig(final long jobId,
+                                                                       final int attempt,
+                                                                       final ConfigType configType,
+                                                                       final JobSyncConfig config,
+                                                                       final StandardSourceDefinition sourceDefinition,
+                                                                       final JsonNode sourceConfiguration)
+      throws IOException {
+    final ConfigReplacer configReplacer = new ConfigReplacer(LOGGER);
+
+    final IntegrationLauncherConfig sourceLauncherConfig = new IntegrationLauncherConfig()
+        .withJobId(String.valueOf(jobId))
+        .withAttemptId((long) attempt)
+        .withDockerImage(config.getSourceDockerImage())
+        .withProtocolVersion(config.getSourceProtocolVersion())
+        .withIsCustomConnector(config.getIsSourceCustomConnector())
+        .withAllowedHosts(configReplacer.getAllowedHosts(sourceDefinition.getAllowedHosts(), sourceConfiguration));
+
+    if (!ConfigType.RESET_CONNECTION.equals(configType)) {
+      sourceLauncherConfig.setAllowedHosts(configReplacer.getAllowedHosts(sourceDefinition.getAllowedHosts(), sourceConfiguration));
+    }
+
+    return sourceLauncherConfig;
+  }
+
+  private IntegrationLauncherConfig getDestinationIntegrationLauncherConfig(final long jobId,
```

Comment:
most of the changes in this file are pulling out some common logic used for both SyncWorkflow and Check input generation.